plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -235,11 +238,13 @@ void chain_plugin::set_program_options(options_description& cli, options_descrip
           "Deferred transactions sent by accounts in this list do not have any of the subjective whitelist/blacklist checks applied to them (may specify multiple times)")
          ("read-mode", boost::program_options::value<eosio::chain::db_read_mode>()->default_value(eosio::chain::db_read_mode::SPECULATIVE),
           "Database read mode (\"speculative\", \"head\", \"read-only\", \"irreversible\").\n"
-          "In \"speculative\" mode database contains changes done up to the head block plus changes made by transactions not yet included to the blockchain.\n"
-          "In \"head\" mode database contains changes done up to the current head block.\n"
-          "In \"read-only\" mode database contains changes done up to the current head block and transactions cannot be pushed to the chain API.\n"
-          "In \"irreversible\" mode database contains changes done up to the last irreversible block and transactions cannot be pushed to the chain API.\n"
+          "In \"speculative\" mode: database contains state changes by transactions in the blockchain up to the head block as well as some transactions not yet included in the blockchain.\n"
+          "In \"head\" mode: database contains state changes by only transactions in the blockchain up to the head block; transactions received by the node are relayed if valid.\n"
+          "In \"read-only\" mode: (DEPRECATED: see p2p-accept-transactions & api-accept-transactions) database contains state changes by only transactions in the blockchain up to the head block; transactions received via the P2P network are not relayed and transactions cannot be pushed via the chain API.\n"
+          "In \"irreversible\" mode: database contains state changes by only transactions in the blockchain up to the last irreversible block; transactions received via the P2P network are not relayed and transactions cannot be pushed via the chain API.\n"
           )
+         ( "p2p-accept-transactions", bpo::value<bool>()->default_value(true), "Allow transactions received over p2p network to be evaluated and relayed if valid.")
```

Comment:
Even though this complicates things, I really think `p2p-accept-transactions` belongs in the net_plugin and not here.

I think what you could do is add another boolean (defaults to false) to the chain_plugin that tracks whether there is any source of transactions (this, or rather its negation, could be accessed via the `no_transactions` accessor). There would then need to be a function that other plugins could call which causes chain_plugin to consider possible other sources of transactions (and update that boolean accordingly). So if `api_accept_transactions` is true, `no_transactions` would always return false. But if `api_accept_transactions` is false, then `no_transactions` would return true by the end of the initialization stage of chain_plugin, but then later switch to false as other plugins (like the net_plugin) finish initialization.

Reply:
Done. Went with `accept_transactions` instead of `no_transactions`